File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets){
  let answer = 0;
  
  targets.sort((a,b) =>  a[1] - b[1]);

  let last = -1;
  for(let [start, end] of targets){
    if(start < last && end >= last){
        continue;
    }
    else{
      answer++;
      last = end;
    }
  }
  return answer;
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java

import java.util.*;
class Solution {
  public int solution(int[][] targets) {
    int answer = 0;

    
    Arrays.sort(targets, (a,b) -> {
      return a[1] - b[1];
    });
    
    int last = -1;
    
    for(int t[] : targets){
      if(t[0] < last && t[1] >= last) continue;
      else{
        last = t[1];
        answer++;
      }
    }
    
    return answer;
  }
}
